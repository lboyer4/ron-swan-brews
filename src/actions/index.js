export const setBreweries = (breweries) => ({
	type: 'SET_BREWERIES',
	breweries
})	