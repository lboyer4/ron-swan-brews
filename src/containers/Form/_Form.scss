form {
    border: 2px solid #A57F60;
    background-color: #F7DBA7;
    opacity: .9;
    margin-left: 20px;
    height: 140px;
    width: 420px;
}


.button-holder {
	display: flex;
	justify-content: space-between;
}

button {
	height: 35px;
 	width: 130px;
}

.show-all,
.show-favorites,
.search {
	width: 130px;
	background-color: #EDBF85;
	border: 1px solid #A57F60;
	font-size: 15px;
	&:hover {
		background-color: #A57F60;
		color: white;
	}
}

label {
	color: black;
	margin: 5px;
}

input {
	width: 390px;
	height: 35px;
	margin: 5px;
}