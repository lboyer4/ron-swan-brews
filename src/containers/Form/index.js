import React, { Component } from 'react';
import './_Form.scss'
import { connect } from 'react-redux';
import { setSearch } from '../../actions';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

export class Form extends Component {
	constructor() {
		super();
		this.state = {
			search: '',
			city: ''
		}
	}

	handleChange = (e) => {
		this.setState({ city: e.target.value})
	}

	handleClick = (e) => {
		e.preventDefault()
		this.setStoreSearch(this.state.city)
	}

	updateSearch = (e) => {
		this.setState({search: e.target.value})
		this.setStoreSearch(e.target.value)
	}

	setStoreSearch = (search) => {
		this.props.setSearch(search)
	}


	render(){
		return(
			<form>
				<label htmlFor="search"> Search Here </label>
				<input 
					type="text" 
					id="search"
					onChange={this.handleChange}
					value={this.state.city}
					name="city"
					placeholder="Search for breweries in a city near you..." />
				<div className="button-holder">
					<input onClick={this.handleClick}
						type="button"
						value="Search"
						className="search"
					/>
					<NavLink to="/show-all">
						<input 
							className="show-all"
							type="button" 
							name="Show All"
							value="Show All" 
							onClick={ this.updateSearch } />
					</NavLink>
					<NavLink to="/favorites">
						<input 
							className="show-favorites"
							type="button" 
							value="Show Favorites" 
							onClick={ this.updateSearch } />
					</NavLink>
				</div>
			</form>
			);
	};
};

export const mapDispatchToProps = (dispatch) => ({
	setSearch: (search) => dispatch(setSearch(search))
});

export default connect(null, mapDispatchToProps)(Form);

Form.propTypes = {
	setSearch: PropTypes.func
}