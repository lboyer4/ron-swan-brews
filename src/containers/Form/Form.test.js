import React from 'react';
import { shallow } from 'enzyme'
import { Form, mapDispatchToProps } from '../Form';
import { setSearch } from '../../actions';

describe('Form', () => {
	let wrapper;
	let mockSetSearch;

	beforeEach(() => {
		mockSetSearch = jest.fn();

		wrapper = shallow(
			<Form 
				setSearch={mockSetSearch}
			/>
		)
	});

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('handleChange', () => {

		it('should set city state when handle change is called', () => {
			const mockCityEvent = {
				target: {
					name: "city",
					value: "Denver"
				}
			}
			wrapper.instance().handleChange(mockCityEvent)
			expect(wrapper.state().city).toEqual("Denver")
		});
	});

	describe('handleClick', () => {

		it('should call setSearch when clicked', () => {
			const mockClickEvent = {
				preventDefault: () => {}
			};
			const mockExpected = 'Denver';
			const mockState = {
				search: 'Denver'
			}
			wrapper.setState(mockState);
			wrapper.instance().handleClick(mockClickEvent);
			expect(wrapper.state().search).toEqual(mockExpected);
		});
	});

	describe('updateSearch', () => {
		it('should set state when called', () => {
			const mockSearchEvent = {
				target: {
					name: "Show All",
					value: "Show All"
				}
			}
			wrapper.instance().updateSearch(mockSearchEvent);
			expect(wrapper.state().search).toEqual("Show All");
		});

		it('should call setSearch', () => {
			const mockSubmitEvent = {
				preventDefault: () => {}
			};
			const mockExpected = 'Show All';
			const mockState = {
				search: 'Show All'
			}
			wrapper.setState(mockState);
			wrapper.instance().handleClick(mockSubmitEvent);
			expect(wrapper.state().search).toEqual(mockExpected);
		});
	});

	describe('setStoreSearch', () => {
		it('should dispatch setSearch', () => {
			const mockSetSearch = jest.fn();
			const mockSearch = {
				search: 'Denver'
			}
			wrapper = shallow(
			<Form 
				search= 'Denver'
				city= 'Denver'
				setSearch={jest.fn()}
			/>
		)
			wrapper.instance().setStoreSearch('Denver')
			expect(wrapper.instance().props.setSearch).toHaveBeenCalledWith('Denver')
		})
	})
	
});