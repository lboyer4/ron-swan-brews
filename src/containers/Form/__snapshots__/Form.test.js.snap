// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
<form>
  <label
    htmlFor="search"
  >
     Search Here 
  </label>
  <input
    id="search"
    name="city"
    onChange={[Function]}
    placeholder="Search for breweries in a city near you..."
    type="text"
    value=""
  />
  <div
    className="button-holder"
  >
    <input
      className="search"
      onClick={[Function]}
      type="button"
      value="Search"
    />
    <NavLink
      to="/show-all"
    >
      <input
        className="show-all"
        name="Show All"
        onClick={[Function]}
        type="button"
        value="Show All"
      />
    </NavLink>
    <NavLink
      to="/favorites"
    >
      <input
        className="show-favorites"
        onClick={[Function]}
        type="button"
        value="Show Favorites"
      />
    </NavLink>
  </div>
</form>
`;
